### PR TITLE
Make error reporting more verbose

### PR DESCRIPTION
**Pull request by alek-sys**
Before only error message was dispalyed in logs providing no information, like exception details or stacktrace. Now error message is logged together with exception, including stacktrace.

### CODE REVIEW

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/deployer/DefaultBackingAppDeploymentService.java [hunk 1]
```diff
@@ -56,7 +56,7 @@ public DefaultBackingAppDeploymentService(DeployerClient deployerClient) {
 			.doOnRequest(l -> log.debug("Updating applications {}", backingApps))
 			.doOnEach(response -> log.debug("Finished updating application {}", response))
 			.doOnComplete(() -> log.debug("Finished updating application {}", backingApps))
-			.doOnError(exception -> log.error("Error updating applications {} with error {}", backingApps, exception));
+			.doOnError(exception -> log.error(String.format("Error updating applications %s with error %s", backingApps, exception)));
```
Comment (Albertoimpl) [inline spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/deployer/DefaultBackingAppDeploymentService.java]: For consistency: `with error '%s'`

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/deployer/DeployerClient.java [hunk 2]
```diff
@@ -67,8 +67,8 @@ public DeployerClient(AppDeployer appDeployer) {
 				.build())
 			.doOnRequest(l -> log.debug("Updating application {}", backingApplication))
 			.doOnSuccess(response -> log.debug("Finished updating application {}", backingApplication))
-			.doOnError(exception -> log.error("Error updating application {} with error {}",
-				backingApplication, exception))
+			.doOnError(exception -> log.error(String.format("Error updating application %s with error %s",
```
Comment (Albertoimpl) [inline spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/deployer/DeployerClient.java]: For consistency: `with error '%s'`

File: spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/service/WorkflowServiceInstanceService.java [hunk 3]
```diff
@@ -143,8 +143,8 @@ public WorkflowServiceInstanceService(ServiceInstanceStateRepository serviceInst
 			.thenMany(invokeDeleteWorkflows(request, response)
 				.doOnRequest(l -> log.debug("Deleting service instance {}", request))
 				.doOnComplete(() -> log.debug("Finished deleting service instance {}", request))
-				.doOnError(exception -> log.error("Error deleting service instance {} with error '{}'",
-					request, exception.getMessage())))
```
Comment (Albertoimpl) [inline spring-cloud-app-broker-core/src/main/java/org/springframework/cloud/appbroker/service/WorkflowServiceInstanceService.java]: for consistency: `%s with error '%s'`

File: spring-cloud-app-broker-security-credhub/src/main/java/org/springframework/cloud/appbroker/workflow/binding/CredHubPersistingCreateServiceInstanceAppBindingWorkflow.java [hunk 4]
```diff
@@ -61,8 +61,8 @@ public CredHubPersistingCreateServiceInstanceAppBindingWorkflow(CredHubOperation
 						.flatMap(credentialName -> persistBindingCredentials(request, response, credentialName)
 							.doOnRequest(l -> LOG.debug("Storing binding credentials with name '{}' in CredHub", credentialName.getName()))
 							.doOnSuccess(r -> LOG.debug("Finished storing binding credentials with name '{}' in CredHub", credentialName.getName()))
-							.doOnError(exception -> LOG.error("Error storing binding credentials with name '{}' in CredHub with error: {}",
-								credentialName.getName(), exception.getMessage())));
+							.doOnError(exception -> LOG.error(String.format("Error storing binding credentials with name '%s' in CredHub with error: %s",
```
Comment (Albertoimpl) [inline spring-cloud-app-broker-security-credhub/src/main/java/org/springframework/cloud/appbroker/workflow/binding/CredHubPersistingCreateServiceInstanceAppBindingWorkflow.java]: for consistency: `%s with error '%s'`

File: spring-cloud-app-broker-security-credhub/src/main/java/org/springframework/cloud/appbroker/workflow/binding/CredHubPersistingDeleteServiceInstanceBindingWorkflow.java [hunk 5]
```diff
@@ -48,8 +48,8 @@ public CredHubPersistingDeleteServiceInstanceBindingWorkflow(CredHubOperations c
 			.flatMap(credentialName -> deleteBindingCredentials(credentialName)
 					.doOnRequest(l -> LOG.debug("Deleting binding credentials with name '{}' in CredHub", credentialName.getName()))
 					.doOnSuccess(r -> LOG.debug("Finished deleting binding credentials with name '{}' in CredHub", credentialName.getName()))
-					.doOnError(exception -> LOG.error("Error deleting binding credentials with name '{}' in CredHub with error: {}",
-						credentialName.getName(), exception.getMessage())))
+					.doOnError(exception -> LOG.error(String.format("Error deleting binding credentials with name '%s' in CredHub with error: %s",
```
Comment (Albertoimpl) [inline spring-cloud-app-broker-security-credhub/src/main/java/org/springframework/cloud/appbroker/workflow/binding/CredHubPersistingDeleteServiceInstanceBindingWorkflow.java]: for consistency: `with error '%s'`

**Comment by alek-sys** [comment]
@Albertoimpl are you ok if I just merge it, despite the failed check? There is no point in writing a test for `String.format` just for coverage.

**Pull request merged**